### ISSUE TITLE
Question: How to generate hyperlinks to worksheets on same workbook

### ISSUE THREAD

**Issue by heumann-a** [dfinke/ImportExcel#426]
Is there a way to set hyperlinks in cells to other Worksheets in the same excel file?

We want to build something like a "Homepage" with Quicklinks to specific places in the workbook instead of searching every time the whole document.

**Comment by jhoneill** [comment]
Yes.  One of the recent versions added processing of hyperlinks. 
If you have text in this format : 
`"xl://internal/sheet1!A1"`
Export-Excel will process it as a hyperlink - if you link to cell A1 it will say "sheet1" as the text, anything else will say "sheet1!Z10" or whatever.
 
I haven't tested linking to a name from this code but I think you can write "`"xl://internal/sheet1!A1"`

If you look at Export-Excel.ps1 , at line 516 you can see how I construct the link and put it in the cell and set the format so it looks like a link, if you have a list of display text and destinations you should be able to re-use that code.

**Comment by dfinke** [comment]
Started to experiment with this. Need to think through how line 516 might be extended so inter-workbook links can handled.

```powershell
$xlFile = "$env:TEMP\testLink.xlsx"
rm $xlFile -ErrorAction Ignore

$map = @{
    A2="sheet2!A1"
    A3="sheet3!A1"
    A4="sheet4!A1"
}

@"
Links
Goto Sheet2
Goto Sheet3
Goto Sheet4
"@ | ConvertFrom-Csv | Export-Excel $xlFile -AutoSize 

"Text`r`nSheet2" | ConvertFrom-Csv | Export-Excel $xlFile -AutoSize -WorksheetName Sheet2
"Text`r`nSheet3" | ConvertFrom-Csv | Export-Excel $xlFile -AutoSize -WorksheetName Sheet3
$pkg = "Text`r`nSheet4" | ConvertFrom-Csv | Export-Excel $xlFile -AutoSize -WorksheetName Sheet4 -PassThru

$ws = $pkg.Workbook.WorkSheets["Sheet1"]

$map.GetEnumerator() | % {
 
    $TargetCell = $ws.Cells[$_.Key]
    $display = $TargetCell.Text
    $referenceAddress = $_.Value

    $h = New-Object -TypeName OfficeOpenXml.ExcelHyperLink -ArgumentList $referenceAddress, $display
    $TargetCell.HyperLink = $h
    $TargetCell.Style.Font.Color.SetColor([System.Drawing.Color]::Blue)
    $TargetCell.Style.Font.UnderLine = $true
}

Close-ExcelPackage $pkg -Show
```

**Comment by jhoneill** [comment]
When I originally wrote it I said "If the data is a hyperlink, then insert it". Then we said "If it is a sting which is valid as a hyperlink then make a hyperlink". 
We might be better having both so that you can create a inter-workbook link with the display text and then that link goes in "as-is".

**Comment by heumann-a** [comment]
Your example worked for me. I only needed to adjust it a little bit.

I have found out, if the Worksheetname have something like a "-" in it, the worksheetnames are additionally single-quoted. To generate succesfully an internal hyperlink it need to looks like this:


```
$map = @{
    A2="**'A-sheet2'**!A1"
    A3="'A-sheet3'!A1"
}

```

thank you for your help!

**Comment by jhoneill** [comment]
Good catch with the quotes. I'll see if I can put a test in for things that need them but don't have them.
